File: app/src/main/java/com/example/githubrepo/services/module/AppModule.java
package com.example.githubrepo.services.module;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import com.example.githubrepo.BaseSearchActivity;
import com.example.githubrepo.MainActivity;
import com.example.githubrepo.MyApplication;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by sarah_neo on 22/02/2017.
 */

@Module
public class AppModule {
    private final MyApplication app;

    public AppModule(MyApplication app) {
        this.app = app;
    }

    @Provides
    @Singleton
    public Application provideApplication() {
        return app;
    }

    @Provides
    @Singleton
    public Context provideApplicationContext() {
        return app;
    }

    @Provides
    @Singleton
    public SharedPreferences provideSharedPrefs() {
        return PreferenceManager.getDefaultSharedPreferences(app);
    }
}

File: app/src/main/java/com/example/githubrepo/services/module/AppModule.kt
package com.example.githubrepo.services.module

import android.app.Application
import android.content.Context
import com.example.githubrepo.BaseSearchActivity
import com.example.githubrepo.MainActivity
import com.example.githubrepo.MyApplication
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module(
        includes = [BusModule::class],
        injects = [
            MyApplication::class,
            BaseSearchActivity::class,
            MainActivity::class
        ],
        library = true
)
class AppModule(private val app: MyApplication) {
    @Provides
    @Singleton
    fun provideApplication(): Application {
        return app
    }

    @Provides
    @Singleton
    fun provideApplicationContext(): Context {
        return app
    }
}